### PR TITLE
Add user common dirs permissions

### PR DESCRIPTION
**Pull request by LaakkonenJussi**
Use includes to add permissions to `Downloads`, `Documents`, `Music`, `Pictures` and `Videos`. In addition include the common user directory `Public` but exclude `Templates` as unneeded and `Desktop` as not wanted our desktop files are handled differently.

Also fix a copy-paste error in `README.md`.

### CODE REVIEW

File: permissions/UserDirs.permission
Comment (Tomin1) [inline permissions/UserDirs.permission:17]: While I'm not totally sure if we should have this directory as a permission, but perhaps, I think in any case it should be a separate permission that is then just included here. That way application can choose to have this permission without access to any of the other directories. The same applies to Templates directory.

Comment (LaakkonenJussi) [reply permissions/UserDirs.permission]: Like 3730ce50bde007216149e30de9010c5f6fc14a3a ?

Comment (Tomin1) [reply permissions/UserDirs.permission]: Yes

Comment (LaakkonenJussi) [reply permissions/UserDirs.permission]: Ok, I'll squash that then to the main commit and wait for final verdict.

Comment (LaakkonenJussi) [reply permissions/UserDirs.permission]: Done.

Comment (Tomin1) [reply permissions/UserDirs.permission]: Oh, sorry, I forgot about the latest change to this: You need to add also noblacklist rule for both directories and then blacklist it in Base.permission ([see the comment at the end](https://github.com/sailfishos/sailjail-permissions/commit/76e8aa5cfb1b279d9a53f1c13842ebc57de812f4#diff-cbb6d3ce8d0bd38db5cb3a5cf84c67aa0c8a7130efb833a164de2613ec01cabfR171)). That way application sees read-only version of the directory unless it has the needed permission.

Comment (LaakkonenJussi) [reply permissions/UserDirs.permission]: Added similar changes with 79cadf1 and updated `README.md` as well.

File: README.md
Comment (spiiroin) [inline README.md:122]: nit: Feel free to ignore. That "(except ..)" portion feels a bit incomplete or something. Maybe  "all except..." or "Desktop is not include because..." or something.

**Review by spiiroin — APPROVED** [review]

File: README.md
Comment (LaakkonenJussi) [reply README.md:122]: Yeah I was thinking if it is reasonable to mention as it is excluded in the list already. Maybe for clarity I just drop that ()-content.

Comment (LaakkonenJussi) [reply README.md]: Dropped the excess content with: 864cd19 - I guess it is better now.

**Review by Tomin1 — APPROVED** [review]
LGTM, remember to squash. :+1:

**Comment by LaakkonenJussi** [comment]
@okodron could you review this as well?

File: permissions/TemplatesDir.permission
Comment (rainemak) [inline permissions/TemplatesDir.permission:11]: I'd not like to promote "Templates" directory (same applies to Desktop which you don't have here). Public is somewhat reasonable.

Comment (rainemak) [reply permissions/TemplatesDir.permission:11]: Not that harmful either but mkdir + blacklist in Base makes sense.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: It is a part of standard Linux desktop. There is a reason to ignore `Desktop` as we handle desktop items differently. There is no real reason to exclude `Templates` especially since if some future partner/client expects it to be there as per to normal XDG user directories, e.g., [Arch ref](https://wiki.archlinux.org/index.php/XDG_user_directories).

File: permissions/Base.permission
Comment (rainemak) [inline permissions/Base.permission:191]: Worth create and blacklist also Desktop. @Tomin1 ?

Comment (LaakkonenJussi) [reply permissions/Base.permission:191]: Makes sense unless we explicitly disallow every dir access by default? I've not been involved in this long enough to understand this completely yet..

File: permissions/TemplatesDir.permission
Comment (rainemak) [reply permissions/TemplatesDir.permission:11]: Correct but intention is to simplify basic use case. If need surfaces later, simple to add.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: Well this is included in the main `UserDirs` permission. I like to think ahead, not to my feet.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: My reasoning here is that if you get the "well known dirs" permission - which effectively means the XDG user directories, you should get the access to all of them that are applicable in our OS.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: And we do have `xdg-user-dir` tool, which [supports these](https://manpages.debian.org/stretch/xdg-user-dirs/xdg-user-dir.1.en.html).

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: Maybe in our case the DESKTOP should actually point in the config to the actual location of desktop files: `$HOME/.local/share/applications` ? So we could include `Desktop` as such and fully support `xdg-user-dir` tool with these permissions as well?

File: permissions/Base.permission
Comment (Tomin1) [reply permissions/Base.permission:191]: Note: We actually have a mention of Desktop in WebView.permission but I'm not sure it should be there and I think we should get rid of that. It's probably not used in practice but again one of those, it happens to access it but doesn't use it so it's there. If we blacklist it here and we keep it there, then there needs to be noblacklist rule.

That said, maybe we should blacklist it to discourage use. By default creating files to wrong places causes them disappear when the sandbox is destructed (application is closed) but this way you can't create files to some of the wrong places which gives developer (or user) immediate feedback instead of wondering what happened.

Comment (LaakkonenJussi) [reply permissions/Base.permission:191]: @Tomin1 or we change the behavior of `XDG_DESKTOP_DIR` to point to where the user app could create additional shortcuts? See https://github.com/sailfishos/sailjail-permissions/pull/62#discussion_r613094062

File: permissions/TemplatesDir.permission
Comment (Tomin1) [reply permissions/TemplatesDir.permission:11]: That same reasoning, that we don't use Desktop directory so it's not needed here, could be applied to Templates as well. We don't use it for anything really and allowing it could make someone think that it's used somewhere. Having Public directory as permission is easier to reason about, it's kind of general place of sharing data between apps when nothing else applies.

I don't think xdg desktop should point to $HOME/.local/share/applications because that's a different thing. It's not desktop directory, it's place to put desktop files that are shown in menu. Although, IIRC, we don't use that either, only the system wide directory.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: It is not a good argument that we do not use it but it exists in standard dirs. If you think a Linux application developer in general and maybe porting of an app to SFOS, then it may be that those dirs are used/required.

`$HOME/.local/share/applications` is not excluded in use but still it works fine and has always worked, I've seen some Harbour apps using it. And I've used it as well to add custom items to our "desktop" view.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: It is just bad to give an assumption that with these permissions you get access to all normal dirs but except you don't. Lets not talk about "well known directories" then. Lets talk about Sailfish OS accepted user home resources?

Comment (rainemak) [reply permissions/TemplatesDir.permission:11]: We are here defining from our point of "well known directories". The UserDirs permission do not even need to be bound to .config/user-dirs.dirs by any means. It's matter of the Permission itself and how do we define it.

File: permissions/Base.permission
Comment (rainemak) [reply permissions/Base.permission:191]: > Note: We actually have a mention of Desktop in WebView.permission but I'm not sure it should be there and I think we should get rid of that. It's probably not used in practice but again one of those, it happens to access it but doesn't use it so it's there. If we blacklist it here and we keep it there, then there needs to be noblacklist rule.

Having "Desktop" in the WebView.permission sounds wrong indeed.

File: permissions/TemplatesDir.permission
Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: @rainemak Um no, simply NO! We are not defining something that is quite commonly accepted to be our own version of something else. I'm not going to be a part of it.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: Lets put it this way, since, for example, `xdg-user-dir` tool is supplied by us it is kind of ridiculous that if application wants `UserDirs` permissions and then can just use a portion of the dirs that are available this tool. It confuses the developer, especially if the dirs are used to retrieve with the tool. One could expect that what that tool provides are usable with the same permissions, right? I see no counter argument to this than "we don't want to" and for me it is not enough.

We should also modify the `Desktop` entry as there is a real argument on not using that. Maybe the .config/user-dirs.dirs should be modified.

File: permissions/Base.permission
Comment (LaakkonenJussi) [reply permissions/Base.permission:191]: Unless we want to give access via that to the `$HOME/.local/share/applications` to add URL desktop entries there to be visible on _our_ desktop?

Comment (LaakkonenJussi) [reply permissions/Base.permission:191]: Eventually resulting in modifying `.config/user-dirs.dirs` as well.

Comment (rainemak) [reply permissions/Base.permission:191]: Correct and that's another task and not related to this per se.

Comment (LaakkonenJussi) [reply permissions/Base.permission:191]: Not related I agree but a worthy issue to raise here as well but not to venture into.

File: permissions/TemplatesDir.permission
Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission:11]: On _normal_ Linux desktop one can put any document there to be used as a template for a new document to be created. Easy to try, just put a document there and use "Create document" anywhere on your system.

There was some talk earlier about an improved office suite made by a partner? We could even utilize/support this ourselves in new document creation. To be following the same line as _normal_ Linux. But if we wish to deviate again...

Comment (pvuorela) [reply permissions/TemplatesDir.permission:11]: In the unlikely event that there would be seen big enough benefit, the question can be always revisited. At the moment we don't even have much support for creating new files (well, happens if you take a picture or download a file, but that's implicit).

For app usage, as also mentioned I can use the template from the file manager, but for doing new file inside the app at least LibreOffice doesn't pick up files from templates at all and uses its own template system. I think that's all for the better and would suggest similar thing for other apps. IMO the whole template directory thing is pretty useless.

Comment (LaakkonenJussi) [reply permissions/TemplatesDir.permission]: Lets deviate then. Changed the PR.

File: permissions/Base.permission
Comment (spiiroin) [reply permissions/Base.permission:193]: NB That "Desktop in WebView.permission" is there due to: initial rules were generated semi-automated and in a mass -> it eliminated an access failure from logs. If it is known to be wrong / lack of it would be inconsequential, it should be removed.

File: README.md
Comment (Tomin1) [inline README.md:121]: Videos too.

**Review by Tomin1 — APPROVED** [review]
Still LGTM

File: permissions/Base.permission
Comment (Thaodan) [reply permissions/Base.permission:193]: > Unless we want to give access via that to the `$HOME/.local/share/applications` to add URL desktop entries there to be visible on _our_ desktop?

I don't think that would make sense since these files should be in the paths which the xdg-menu spec specifies:
https://specifications.freedesktop.org/menu-spec/menu-spec-latest.html#paths

Also the intend behind this is different IMHO. It is more a place to put desktop files there that are already in the menu or link common folder etc.

If we want to support adding new desktop entries that would be another task see:
https://specifications.freedesktop.org/menu-spec/menu-spec-latest.html#menu-editing

**Comment by LaakkonenJussi** [comment]
@okodron Still waiting for approval from OMP side.

**Review by okodron — APPROVED** [review]

**Pull request merged**
